WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+            if (copiedSet == updatedPluginSlugSet) {
+                updateAllPluginListsWithNewPlugins(copiedSet)
+                updatedPluginSlugSet.clear()
+            }
+        }, 250)
+    }
+
+    private fun updateAllPluginListsWithNewPlugins(updatedPluginSlugSet: Set<String>) {
+        if (updatedPluginSlugSet.isEmpty()) {
+            return
+        }
+        val newPluginMap = HashMap<String, ImmutablePluginModel>(updatedPluginSlugSet.size)
+        for (slug in updatedPluginSlugSet) {
+            val immutablePlugin = mPluginStore.getImmutablePluginBySlug(site!!, slug)
+            if (immutablePlugin != null) {
+                newPluginMap[slug] = immutablePlugin
+            }
+        }
+        // By combining all the updated plugins into one map, we can post a single update to the UI after changes are
+        // reflected
+        updatePluginListWithNewPlugin(_featuredPlugins, newPluginMap)
+        updatePluginListWithNewPlugin(_newPlugins, newPluginMap)
+        updatePluginListWithNewPlugin(_popularPlugins, newPluginMap)
+        updatePluginListWithNewPlugin(_searchResults, newPluginMap)
+
+        // Unfortunately we can't use the same method to update the site plugins because removing/installing plugins can
+        // mess up the list. Also we care most about the Site Plugins and using the store to get the correct plugin
+        // information is much more reliable than any manual update we can make
+        reloadPluginDirectory(PluginDirectoryType.SITE)
+    }
+
+    private fun updatePluginListWithNewPlugin(mutableLiveData: MutableLiveData<List<ImmutablePluginModel>>,
+                                              newPluginMap: Map<String, ImmutablePluginModel>) {
+        val pluginList = mutableLiveData.value
+        if (pluginList == null || pluginList.isEmpty() || newPluginMap.isEmpty()) {
+            // Nothing to update
+            return
+        }
+        // When a site or wporg plugin is updated we need to update every occurrence of that item
+        val newList = ArrayList<ImmutablePluginModel>(pluginList.size)
+        var isChanged = false
+        for (immutablePlugin in pluginList) {
+            val slug = immutablePlugin.slug
+            val newPlugin = newPluginMap[slug]
+            if (newPlugin != null) {
+                // add new item
+                newList.add(newPlugin)
+                isChanged = true
+            } else {
+                // add old item
+                newList.add(immutablePlugin)
+            }
+        }
+        // Only update if the list is actually changed
+        if (isChanged) {
+            mutableLiveData.postValue(newList)
+        }
+    }
+
+    private fun shouldSearch(): Boolean {
+        // We need at least 2 characters to be able to search plugins
+        return searchQuery.length > 1
+    }
+
+    private fun submitSearch(query: String?, delayed: Boolean) {
```

Comment:
I think `query` can be non-nullable here.

Reply:
🤦‍♂️ I even handled the `null` case in the activity and forgot to make this non-nullable. Fixed in fd11233d1c822c64957356fff25f5ecacc0cfcec.